masonry_core/src/core/contexts.rs

```diff
@@ -1128,6 +1128,11 @@ impl_context_method!(MutateCtx<'_>, EventCtx<'_>, UpdateCtx<'_>, {
             .expect("remove_child: child not found");
         self.global_state.scenes.remove(&child.id());
 
+        // If we remove the focus anchor, its parent becomes the anchor.
+        if self.global_state.focus_anchor == Some(id) {
+            self.global_state.focus_anchor = Some(self.widget_state.id);
+        }
```

Comment:
What happens if the focus anchor isn't the direct child, but a descendant of that child? Do we revert to the root?